### PR TITLE
[python] Set PIP_USER=no

### PR DESCRIPTION
**Pull request by csweichel**
fixes https://github.com/gitpod-io/gitpod/issues/4886

### CODE REVIEW

**Review by aledbf — APPROVED** [review]
LGTM

**Comment by aledbf** [comment]
@csweichel let's merge https://github.com/gitpod-io/workspace-images/pull/458 before this one, cancel the circle CI build, and merge this PR to release the new images?

**Comment by jankeromnes** [comment]
@csweichel @aledbf Many thanks! 💯

@AlexTugarev Since you previously worked on the Python setup, please have a look as well if you have time. 🙂

Also, on a more general note, I believe we should soon do two things with our Python setup:

- Drop support for (deprecated) Python 2, and in the same run maybe remove Pyenv entirely (and keep only Ubuntu's stock Python) -- relatedly, now that we have Full Workspace Backup generally enabled, I think we can remove a ton of `/workspace` hacks

- Compare our Python stack with https://github.com/microsoft/vscode-dev-containers/blob/main/script-library/python-debian.sh and see if maybe there are a few things we could do better (e.g. some additional tools to install, or if there exists a more common / saner Python dev setup)

File: full/Dockerfile
Comment (AlexTugarev) [inline full/Dockerfile:157]: why unsetting it here whereas its set in [gitpod-layer](https://github.com/gitpod-io/gitpod/blob/24f7b609bf79b247d72d1841282639726bcd5891/components/image-builder/workspace-image-layer/gitpod-layer/alpine/gitpod/.bashrc-append#L11) on top? my first impression is that this won't be effective. 

also, just a reminder, without forcibly enabling `PYTHONUSERBASE` under `/workspace` the backup strategy for python is gone, and prebuilds as well. using a virutal env setup seems to be the best way. it would be great to provide a replacement that works.

**Comment by AlexTugarev** [comment]
> ... now that we have Full Workspace Backup generally enabled ...

@jankeromnes, we do?

**Comment by jankeromnes** [comment]
> > ... now that we have Full Workspace Backup generally enabled ...
> 
> @jankeromnes, we do?

We don't? 😳 @csweichel @aledbf do you know?

**Comment by aledbf** [comment]
> We don't? flushed @csweichel @aledbf do you know?

It is not enabled yet (there is an issue with the user permissions when we create the workspace backup)

**Review by jankeromnes — CHANGES_REQUESTED** [review]
Thanks again for improving Gitpod's Python setup! 💯

Blocking this PR until https://github.com/gitpod-io/workspace-images/pull/461#discussion_r680693077 is resolved, so that we don't accidentally merge it as is.

File: full/Dockerfile
Comment (aledbf) [reply full/Dockerfile:157]: @AlexTugarev the gitpod-layer was removed https://github.com/gitpod-io/gitpod/pull/4923. I will add the `.bashrc-append` changes here.

**Comment by aledbf** [comment]
@csweichel @jankeromnes @AlexTugarev can we merge this PR? Without it, most of the examples are broken when the new image builder mk3 is used (due to the removal of the gitpod layer)
(needs a rebase first)

File: full/Dockerfile
Comment (jankeromnes) [inline full/Dockerfile:181]: I think this does more harm than good. Maybe let's remove it for now?

```suggestion
```

File: full/Dockerfile
Comment (jankeromnes) [inline full/Dockerfile:160]: I have no Python experience to tell whether setting any of these variables is okay.

However, I don't think you'll find any reviewers here with Python experience, so continuing to wait for a review here is unhelpful.

Also, if possible, I would very much prefer not setting any of these variables. I'm aware that this may break package persistence (they'll land outside of `/workspace`) but for that I'd rather focus on enabling Full Workspace Backup instead.

File: full/Dockerfile
Comment (jankeromnes) [inline full/Dockerfile:205]: Can be simplified. Also, shouldn't it be `gitpod:gitpod` instead of `gitpod.gitpod`?

```suggestion
RUN sudo mkdir -p $CARGO_HOME \
    && sudo chown -R gitpod:gitpod $CARGO_HOME
```

**Review by jankeromnes — APPROVED** [review]
Approving to unblock this PR, but I would very very much prefer it if we remove all of these silly variables.

- For Python, we're unwittingly breaking many people's setups by forcing some PIP_USER or VENV things

- For Ruby, our setup is so broken that every major Ruby project just replaces it entirely in a Dockerfile (also, we use RVM instead of today's standard Rbenv)

- For Rust, the /workspace hack still looks okay, but here too I'd very much prefer removing the hack altogether, and instead enable Full Workspace Backup everywhere

File: full/Dockerfile
Comment (aledbf) [reply full/Dockerfile]: done

File: full/Dockerfile
Comment (aledbf) [reply full/Dockerfile:160]: >  but for that I'd rather focus on enabling Full Workspace Backup instead.

this is one of those issues where we have a circular dependency; we need this PR to switch to the new image builder mk3. With that, we can focus on the dazzle refactor that will reduce the image size and simplify the dependencies tangle, and then we can start using stargz. After that, we should focus on Full Workspace Backup. Without all those previous steps we are not improving the size, usability, or modularity of the workspace container images.

**Pull request merged**
